app/page.tsx

```diff
@@ -1,15 +1,21 @@
 "use client";
-
-import SearchBar from "@/components/SearchBar";
 import React from "react";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import SearchBar from "@/components/SearchBar";
+import SearchResult from "../components/SearchResult";
+
+const queryClient = new QueryClient();
 
 export default function Home() {
   const [searchValue, setSearchValue] = React.useState<string>("");
 
   return (
-    <main className="flex flex-col items-center justify-center ">
-      <SearchBar onChange={(value) => setSearchValue(value)} />
-      <h1 className="font-bold text-6xl mt-10">Home Page</h1>
-    </main>
+    <QueryClientProvider client={queryClient}>
```

Comment:
this is a react context provider btw, don't think you've learned about this yet.

I'd watch a video to get a high-level overview, [or even try reading the docs on it.](https://react.dev/learn/passing-data-deeply-with-context) The new react docs are super good, it are the best/most accurate mental model you're going to find on anything react.

react-query uses context (so you won't directly be using the `useContext` hook, but it's good to know what it generally is and how to use it)

Reply:
Read the docs, super useful information